internal/controller/services/secretgenerator/secret_test.go

```diff
@@ -14,7 +18,85 @@ const (
 	errUnsupportedType        = "secret type is not supported"
 )
 
+var (
+	base64Regex = regexp.MustCompile("^[A-Za-z0-9+/]*={0,3}$")
+)
+
 func TestNewSecret(t *testing.T) {
+	tests := []struct {
+		name               string
+		secretName         string
+		secretType         string
+		complexity         int
+		expectedResult     string
+		expectedErrMessage string
+	}{
+		{
+			name:           "random case",
+			secretName:     "my-secret",
+			secretType:     "random",
+			complexity:     1,
+			expectedResult: "success",
+		},
+		{
+			name:           "oauth case",
+			secretName:     "another-secret",
+			secretType:     "oauth",
+			complexity:     1,
+			expectedResult: "success",
+		},
+		{
+			name:               "unsupported secret type",
+			secretName:         "my-secret",
+			secretType:         "·%$@&?",
+			complexity:         1,
+			expectedResult:     "error",
+			expectedErrMessage: errUnsupportedType,
+		},
+		{
+			name:           "zero complexity",
+			secretName:     "my-secret",
+			secretType:     "random",
+			complexity:     0,
+			expectedResult: "nil",
+		},
+		{
+			name:           "empty name",
+			secretName:     "",
+			secretType:     "random",
+			complexity:     1,
+			expectedResult: "success",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			secret, err := secretgenerator.NewSecret(tt.secretName, tt.secretType, tt.complexity)
+			switch tt.expectedResult {
+			case "error":
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), tt.expectedErrMessage)
+				return // Early return after error validation
+			case "nil":
+				require.NoError(t, err)
+				return // Early return for nil case
+			case "success":
+				require.NoError(t, err)
+				require.NotNil(t, secret)
+				assert.Equal(t, tt.secretName, secret.Name)
+				assert.Equal(t, tt.secretType, secret.Type)
+				assert.Equal(t, tt.complexity, secret.Complexity)
+				assert.NotEmpty(t, secret.Value)
+				assert.True(t, base64Regex.MatchString(secret.Value), "Secret value should be base64 encoded")
+			default:
+				assert.Empty(t, secret.Value)
```

Comment:
I guess this is not needed to be checked as it means the test case doesn't have a supported expectedResult value.

Reply:
It was useful during test development so to make sure there were no typos but I can remove it